corehq/util/dates.py

```diff
@@ -18,3 +18,28 @@ def get_timestamp(date):
 
 def get_timestamp_millis(date):
     return 1000 * get_timestamp(date)
+
+
+def safe_strftime(val, fmt):
```

Comment:
Interesting, definitely and edge case. Could also just check the the year is pre 1900 and if not proceed normally.